core/src/main/scala/cats/instances/SymbolInstances.scala

```diff
@@ -25,5 +25,5 @@ import cats.Show
 
 trait SymbolInstances extends cats.kernel.instances.SymbolInstances {
   implicit val catsStdShowForSymbol: Show[Symbol] =
-    Show.fromToString[Symbol]
+    Show.show("'" + _.name)
```

Comment:
Or maybe this should just be `.name` - without the leading `'`? I think of `Show` as being something nice / sane for humans to read - we have `.toString` for diagnostic / debug output...

Reply:
I think this isn't safe since Symbol("foo bar") is allowed I think.

Instead I think we should check if the name is a valid Symbol literal (somehow, maybe regex?) and then we could use the '

Reply:
Not understanding what the issue is - what does being a valid 2.12 Symbol literal have to do with the `Show` representation? Is there a restriction that `Show` has to emit valid Scala source code? It already breaks that on the `Show` representation for String. `'foo bar` is a valid string. That's the `.toString` of `Symbol` in Scala 2.12.

Reply:
This is literally already the current `Show[Symbol]` impl on 2.12. Having different goals for `Show` seems outside the scope of this PR. I don't care what it is, so long as it's consistent. Do you have any alternate suggestions? 2.13's `Symbol(foo)`? Or no decoration, just `foo`?

Reply:
As far as the `Show` instance for `Symbol`, would you prefer:
* `'foo bar` (the current impl for 2.12)
* `Symbol(foo bar)` (the current impl for 2.13)
* something else?

The only thing his PR is trying to do is to make that impl consistent across Scala versions. I'm not interested in redesigning / removing the `Show` typeclass on this PR.

Reply:
Related (there's no Read typeclass - Show is for human-readable output, not intended to be machine-parseable): https://github.com/typelevel/cats/issues/932#issuecomment-196825443